### PR TITLE
Add transaction error tracking section

### PR DESCRIPTION
**Pull request by igneel64**
Added the tracking transaction errors UI part.

Additionally includes a server side example using Node.js. At some point we can include any other server-side tracker example.

### CODE REVIEW

File: content/tracking/transaction_error.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
++++
+title = "Track Transaction Errors"
+weight = 8
+post = ""
++++
+
+Transaction error events can be used to track issues during any step of a transaction being processed by the system. Even if the API is generic and can be used in any case of a transaction not being completed, there is a plethora of payment processor/gateway errors which are extremely useful to include in your warehouse together with the rich Snowplow schema such as:
```
Comment (Jack-Keene) [inline content/tracking/transaction_error.md:7]:
```suggestion
Transaction error events can be used to track issues during any step of a transaction being processed by the system. The API is generic and can be used in any case of an incomplete transaction, including payment processor/gateway errors. These data can be combined with the rich Snowplow schema to provide insights such as:
```
Slight wording change

File: content/tracking/transaction_error.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+- Fraud rejections and other risk related rejections
+- Processor/gateway internal errors
+
+Additionally by tracking the error types, soft or hard and the resolution selected by your system, you can test and optimize the flows and experience you provide to your customers facing issues with completing a transaction.
```
Comment (Jack-Keene) [inline content/tracking/transaction_error.md:11]:
```suggestion
Additionally by tracking the error types, soft or hard, and the resolution selected by your system, you can test and optimize the flows and experience you provide to your customers facing issues with completing a transaction.
```
Comma

File: content/tracking/transaction_error.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  }
+});
+```
+
+{{% /tab %}}
+
+{{% tab name="Node.js" %}}
+#### Track transaction errors using a self describing event
+
+To track transaction errors in the Node.js tracker, you can use the `buildSelfDescribingEvents` method together with the correct parameters for the transaction error event.
```
Comment (Jack-Keene) [inline content/tracking/transaction_error.md:116]:
```suggestion
To track transaction errors in the Node.js tracker, you can use the `buildSelfDescribingEvent` method together with the correct parameters for the transaction error event.
```
I think the method is `buildSelfDescribingEvent`, if not it's wrong in the code snippet

File: content/tracking/transaction_error.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+      revenue: 45,
+      currency: "EUR",
+      transaction_id: "T12345",
+      payment_method: "card" 
+    }
+  }
+]));
+```
+
+To learn more about tracking self describing events in Node.js take a look at the `buildSelfDescribingEvents` [documentation](https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/javascript-trackers/node-js-tracker/node-js-tracker-v3/tracking-events/#track-self-describing-events-withbuildselfdescribingevent).
```
Comment (Jack-Keene) [inline content/tracking/transaction_error.md:191]:
```suggestion
To learn more about tracking self describing events in Node.js take a look at the `buildSelfDescribingEvent` [documentation](https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/javascript-trackers/node-js-tracker/node-js-tracker-v3/tracking-events/#track-self-describing-events-withbuildselfdescribingevent).
```

**Review by Jack-Keene — APPROVED** [review]
LGTM, just a few minor changes!

File: content/tracking/transaction_error.md (hunk 3 above)
Comment (igneel64) [reply content/tracking/transaction_error.md]: Yeap all fixed, thanks!

**Pull request merged**